app/views/admissions_admin/admissions/show.html.haml

```diff
@@ -17,11 +17,28 @@
 %h1.mt-3.mb-2
   = set_and_return_title T('admissions.my_groups')
 
-.flex-row.wrap
-  - @my_groups.sort.each do |group|
-    = link_to admissions_admin_admission_group_path(@admission, group) do
-      .samf-button.plain.small.mb-2.mr-2
-        = group.name
+%div
   - if @my_groups.empty?
     %li
       %p= t('admissions.no_groups')
+  - elsif @my_groups.count < 5
+    -# Dont group by letter if < 5
+    .flex-row.wrap
+      - @my_groups.sort.each do |g|
+        = link_to admissions_admin_admission_group_path(@admission, g) do
+          .samf-button.plain.small.mb-1.mt-1.mr-2{style: "float: left;"}
```

Comment:
Inline CSS that could be moved to the respective stylesheet (and some other places in this PR).